src/com/amazon/corretto/crypto/provider/AesGcmSpi.java

```diff
@@ -738,12 +738,9 @@ private int engineDecryptFinal(
         }
       }
       // Decryption completed successfully.
+      // Copy from working buffer into actual output
+      System.arraycopy(workingInputArray, workingInputOffset, output, outputOffset, outLen);
```

Comment:
Maybe we can avoid this copy in case the output buffer doesn't overlap input. Still in the case of failure the output buffer would potentially be modified, but it's output buffer after all.